examples/medplum-scheduling-demo/src/bots/core/cancel-appointment.ts

```diff
@@ -0,0 +1,30 @@
+import { BotEvent, MedplumClient, resolveId } from '@medplum/core';
+import { Appointment, Slot } from '@medplum/fhirtypes';
+
+export async function handler(medplum: MedplumClient, event: BotEvent<Appointment>): Promise<Appointment> {
```

Comment:
The bot implements a simplified logic. Let me know if that's enough for the demo app.

Reply:
One small change: Can you implement both updates inside a `Bundle` with type transaction. We'll very soon have better transactional semantics, meaning that the writes will succeed or fail together

Reply:
Sure, done